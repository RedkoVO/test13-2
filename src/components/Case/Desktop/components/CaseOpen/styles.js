import BackArrow from '../../../../../assets/images/svg/Backarrow.svg'
import CaseStar from '../../../../../assets/images/svg/Casestar.svg'

export default () => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    width: '800px',
    marginTop: '25px',
    minHeight: '600px',
    borderRadius: '20px',
    backgroundColor: 'rgba(0,0,0,0.13)'
  },
  caseOpenTop: {
    position: 'relative',
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    width: '100%',
    height: '200px',
    padding: '30px 35px',
    paddingBottom: 0,
    boxSizing: 'border-box',
    justifyContent: 'center'
  },
  caseOpenBackBtn: {
    position: 'absolute',
    alignSelf: 'flex-start',
    width: '22px',
    height: '22px',
    background: `url(${BackArrow})`,
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'center',
    backgroundSize: 'cover',
    top: '30px',
    left: '35px'
  },
  caseOpenTopImg: {
    width: '250px',
    height: '170px',

    '& img': {
      maxWidth: '100%'
    }
  },
  caseOpenTopDescr: {
    position: 'relative',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    width: '160px',
    paddingTop: '15px',
    boxSizing: 'border-box'
  },
  caseOpenTopMarker: {
    position: 'absolute',
    width: '15px',
    height: '15px',
    background: `url(${CaseStar})`,
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'center',
    backgroundSize: 'cover',
    right: '0px',
    top: 0
  },
  caseOpenTopName: {
    fontSize: '15px',

    '&:span': {
      fontWeight: 500
    }
  },
  caseOpenTopPrice: {
    fontSize: '12px',
    marginTop: '8px',
    marginBottom: '5px',
    padding: '5px 15px',
    borderRadius: '15px',
    backgroundColor: 'rgba(0,0,0,0.13)',

    '&:span': {
      fontWeight: 500
    }
  },
  caseOpenTopGame: {
    fontSize: '11px',

    '&:span': {
      fontWeight: 500
    }
  },
  caseOpenContent: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    flexWrap: 'wrap',
    width: '100%',
    padding: '30px 25px 30px 35px',
    boxSizing: 'border-box'
  },
  caseOpenTitle: {
    flexBasis: '100%',
    textAlign: 'center',
    fontSize: '17px',
    fontWeight: 500,
    marginBottom: '25px'
  },
  caseOpenItem: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    width: '100px',
    minHeight: '100px',
    backgroundColor: 'rgba(0,0,0,0.13)',
    borderRadius: '15px',
    marginRight: '10px',
    marginBottom: '10px',
    padding: '5px',
    boxSizing: 'border-box'
  },
  caseOpenItemImg: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    width: '95px',
    height: '55px',
    marginBottom: '10px',

    '& img': {
      maxWidth: '100%',
      maxHeight: '100%'
    }
  },
  caseOpenItemName: {
    fontSize: '10px',
    textAlign: 'center',

    '&span': {
      fontWeight: 700
    }
  }
})
