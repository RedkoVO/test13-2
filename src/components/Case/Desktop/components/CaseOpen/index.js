import React from 'react'
import PropTypes from 'prop-types'
import withStyles from '@material-ui/core/styles/withStyles'

import styles from './styles'

import CaseXOpen from '../../../../../assets/images/caseXOpen.png'

const CaseOpen = ({ classes, caseItems }) => (
  <div className={classes.root}>
    <div className={classes.caseOpenTop}>
      <div className={classes.caseOpenBackBtn} />
      <div className={classes.caseOpenTopImg}>
        <img src={CaseXOpen} alt="" />
      </div>
      <div className={classes.caseOpenTopDescr}>
        <div className={classes.caseOpenTopMarker} />
        <div className={classes.caseOpenTopName}>
          Case: <span>Hello, buddy!</span>
        </div>
        <div className={classes.caseOpenTopPrice}>
          Open for <span>$34.20</span>
        </div>
        <div className={classes.caseOpenTopGame}>
          Case: <span>CS:GO</span>
        </div>
      </div>
    </div>
    <div className={classes.caseOpenContent}>
      <div className={classes.caseOpenTitle}>This case cointains this stuff:</div>

      {caseItems.map(item => (
        <div className={classes.caseOpenItem} key={item.id}>
          <div className={classes.caseOpenItemImg}>
            <img src={item.img} alt="" />
          </div>
          <div className={classes.caseOpenItemName}>
            Gun: <span>{item.title}</span>
          </div>
        </div>
      ))}
    </div>
  </div>
)

CaseOpen.propTypes = {
  classes: PropTypes.object,
  caseItems: PropTypes.array
}

export default withStyles(styles)(CaseOpen)
