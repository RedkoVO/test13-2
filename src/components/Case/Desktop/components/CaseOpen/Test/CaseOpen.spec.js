import * as React from 'react'
import renderer from 'react-test-renderer'

import CaseOpen from '../'

import GunCase from '../../../../../../assets/images/GunCase.png'

const caseItems = [
  { id: 1, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 2, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 3, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 4, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 5, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 6, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 7, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 8, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 9, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 10, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 11, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 12, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 13, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 14, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 15, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 16, title: 'Good Weapon2 1.2', img: GunCase }
]

it('renders correctly', () => {
  const tree = renderer.create(<CaseOpen caseItems={caseItems} />).toJSON()
  expect(tree).toMatchSnapshot()
})
