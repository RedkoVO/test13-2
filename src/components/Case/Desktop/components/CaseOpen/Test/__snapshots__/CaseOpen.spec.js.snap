// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="CaseOpen-root-1"
>
  <div
    className="CaseOpen-caseOpenTop-2"
  >
    <div
      className="CaseOpen-caseOpenBackBtn-3"
    />
    <div
      className="CaseOpen-caseOpenTopImg-4"
    >
      <img
        alt=""
        src="caseXOpen.png"
      />
    </div>
    <div
      className="CaseOpen-caseOpenTopDescr-5"
    >
      <div
        className="CaseOpen-caseOpenTopMarker-6"
      />
      <div
        className="CaseOpen-caseOpenTopName-7"
      >
        Case: 
        <span>
          Hello, buddy!
        </span>
      </div>
      <div
        className="CaseOpen-caseOpenTopPrice-8"
      >
        Open for 
        <span>
          $34.20
        </span>
      </div>
      <div
        className="CaseOpen-caseOpenTopGame-9"
      >
        Case: 
        <span>
          CS:GO
        </span>
      </div>
    </div>
  </div>
  <div
    className="CaseOpen-caseOpenContent-10"
  >
    <div
      className="CaseOpen-caseOpenTitle-11"
    >
      This case cointains this stuff:
    </div>
    <div
      className="CaseOpen-caseOpenItem-12"
    >
      <div
        className="CaseOpen-caseOpenItemImg-13"
      >
        <img
          alt=""
          src="GunCase.png"
        />
      </div>
      <div
        className="CaseOpen-caseOpenItemName-14"
      >
        Gun: 
        <span>
          Good Weapon2 1.2
        </span>
      </div>
    </div>
    <div
      className="CaseOpen-caseOpenItem-12"
    >
      <div
        className="CaseOpen-caseOpenItemImg-13"
      >
        <img
          alt=""
          src="GunCase.png"
        />
      </div>
      <div
        className="CaseOpen-caseOpenItemName-14"
      >
        Gun: 
        <span>
          Good Weapon2 1.2
        </span>
      </div>
    </div>
    <div
      className="CaseOpen-caseOpenItem-12"
    >
      <div
        className="CaseOpen-caseOpenItemImg-13"
      >
        <img
          alt=""
          src="GunCase.png"
        />
      </div>
      <div
        className="CaseOpen-caseOpenItemName-14"
      >
        Gun: 
        <span>
          Good Weapon2 1.2
        </span>
      </div>
    </div>
    <div
      className="CaseOpen-caseOpenItem-12"
    >
      <div
        className="CaseOpen-caseOpenItemImg-13"
      >
        <img
          alt=""
          src="GunCase.png"
        />
      </div>
      <div
        className="CaseOpen-caseOpenItemName-14"
      >
        Gun: 
        <span>
          Good Weapon2 1.2
        </span>
      </div>
    </div>
    <div
      className="CaseOpen-caseOpenItem-12"
    >
      <div
        className="CaseOpen-caseOpenItemImg-13"
      >
        <img
          alt=""
          src="GunCase.png"
        />
      </div>
      <div
        className="CaseOpen-caseOpenItemName-14"
      >
        Gun: 
        <span>
          Good Weapon2 1.2
        </span>
      </div>
    </div>
    <div
      className="CaseOpen-caseOpenItem-12"
    >
      <div
        className="CaseOpen-caseOpenItemImg-13"
      >
        <img
          alt=""
          src="GunCase.png"
        />
      </div>
      <div
        className="CaseOpen-caseOpenItemName-14"
      >
        Gun: 
        <span>
          Good Weapon2 1.2
        </span>
      </div>
    </div>
    <div
      className="CaseOpen-caseOpenItem-12"
    >
      <div
        className="CaseOpen-caseOpenItemImg-13"
      >
        <img
          alt=""
          src="GunCase.png"
        />
      </div>
      <div
        className="CaseOpen-caseOpenItemName-14"
      >
        Gun: 
        <span>
          Good Weapon2 1.2
        </span>
      </div>
    </div>
    <div
      className="CaseOpen-caseOpenItem-12"
    >
      <div
        className="CaseOpen-caseOpenItemImg-13"
      >
        <img
          alt=""
          src="GunCase.png"
        />
      </div>
      <div
        className="CaseOpen-caseOpenItemName-14"
      >
        Gun: 
        <span>
          Good Weapon2 1.2
        </span>
      </div>
    </div>
    <div
      className="CaseOpen-caseOpenItem-12"
    >
      <div
        className="CaseOpen-caseOpenItemImg-13"
      >
        <img
          alt=""
          src="GunCase.png"
        />
      </div>
      <div
        className="CaseOpen-caseOpenItemName-14"
      >
        Gun: 
        <span>
          Good Weapon2 1.2
        </span>
      </div>
    </div>
    <div
      className="CaseOpen-caseOpenItem-12"
    >
      <div
        className="CaseOpen-caseOpenItemImg-13"
      >
        <img
          alt=""
          src="GunCase.png"
        />
      </div>
      <div
        className="CaseOpen-caseOpenItemName-14"
      >
        Gun: 
        <span>
          Good Weapon2 1.2
        </span>
      </div>
    </div>
    <div
      className="CaseOpen-caseOpenItem-12"
    >
      <div
        className="CaseOpen-caseOpenItemImg-13"
      >
        <img
          alt=""
          src="GunCase.png"
        />
      </div>
      <div
        className="CaseOpen-caseOpenItemName-14"
      >
        Gun: 
        <span>
          Good Weapon2 1.2
        </span>
      </div>
    </div>
    <div
      className="CaseOpen-caseOpenItem-12"
    >
      <div
        className="CaseOpen-caseOpenItemImg-13"
      >
        <img
          alt=""
          src="GunCase.png"
        />
      </div>
      <div
        className="CaseOpen-caseOpenItemName-14"
      >
        Gun: 
        <span>
          Good Weapon2 1.2
        </span>
      </div>
    </div>
    <div
      className="CaseOpen-caseOpenItem-12"
    >
      <div
        className="CaseOpen-caseOpenItemImg-13"
      >
        <img
          alt=""
          src="GunCase.png"
        />
      </div>
      <div
        className="CaseOpen-caseOpenItemName-14"
      >
        Gun: 
        <span>
          Good Weapon2 1.2
        </span>
      </div>
    </div>
    <div
      className="CaseOpen-caseOpenItem-12"
    >
      <div
        className="CaseOpen-caseOpenItemImg-13"
      >
        <img
          alt=""
          src="GunCase.png"
        />
      </div>
      <div
        className="CaseOpen-caseOpenItemName-14"
      >
        Gun: 
        <span>
          Good Weapon2 1.2
        </span>
      </div>
    </div>
    <div
      className="CaseOpen-caseOpenItem-12"
    >
      <div
        className="CaseOpen-caseOpenItemImg-13"
      >
        <img
          alt=""
          src="GunCase.png"
        />
      </div>
      <div
        className="CaseOpen-caseOpenItemName-14"
      >
        Gun: 
        <span>
          Good Weapon2 1.2
        </span>
      </div>
    </div>
    <div
      className="CaseOpen-caseOpenItem-12"
    >
      <div
        className="CaseOpen-caseOpenItemImg-13"
      >
        <img
          alt=""
          src="GunCase.png"
        />
      </div>
      <div
        className="CaseOpen-caseOpenItemName-14"
      >
        Gun: 
        <span>
          Good Weapon2 1.2
        </span>
      </div>
    </div>
  </div>
</div>
`;
